File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    ls = []
    for i in range(left,right+1):
        if (len([x for x in range(1,i//2+1) if i%x==0])+1) % 2 == 0:
            ls.append(i)
        else:
            ls.append(-i)
    return sum(ls)

    

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int sum = 0;
        for (int i = left; i <= right; i++) {
            if (countDivisor(i) % 2 == 0) sum += i;
            else sum -= i;
        }
        return sum;
    }
    
    public static int countDivisor(int n) {
        int count = 0;
        for (int i = 1; i * i <= n; i++) {
            if (i * i == n) count++;
            else if (n % i == 0) count += 2;
        }
        return count;
    }
}